### ISSUE TITLE
Release 3.0.1 does not send "last will"

### ISSUE THREAD

**Issue by vborcea** [emqx/emqx#2274]
#### Environment

- OS: image: emqx/emqx
#### Description

Hello,

We have a cluster deployed using image 3.0.1 and it doesn't send the "last will" message. Please note that with 3.0.0 there was no problem.

```
apiVersion: apps/v1beta1
kind: StatefulSet
metadata:
  name: emqtt
  namespace: default
spec:
  replicas: 3
  serviceName: emqtt
  template:
    metadata:
      labels:
        app: emqtt
    spec:
      serviceAccountName: emqtt
      containers:
        - name: emqtt
          image: emqx/emqx:v3.0.1
          ports:
            - containerPort: 1883
            - containerPort: 8883
            - containerPort: 8080
            - containerPort: 18083
            - containerPort: 4369
            - containerPort: 4370
            - containerPort: 6369
            - containerPort: 6370
            - containerPort: 6371
            - containerPort: 6372
            - containerPort: 6373
            - containerPort: 6374
            - containerPort: 6375
            - containerPort: 6376
            - containerPort: 6377
            - containerPort: 6378
          env:
            - name: EMQX_NAME
              value: "emqtt"
            - name: EMQX_CLUSTER__DISCOVERY
              value: "k8s"
            - name: EMQX_CLUSTER__K8S__NAMESPACE
              value: "default"
            - name: EMQX_CLUSTER__K8S__APISERVER
              value: https://kubernetes.default.svc.cluster.local:443
            - name: EMQX_CLUSTER__K8S__SERVICE_NAME
              value: "emqtt"
            - name: EMQX_CLUSTER__K8S__ADDRESS_TYPE
              value: "ip"
            - name: EMQX_CLUSTER__K8S__APP_NAME
              value: "emqtt"
            - name: EMQX_LOADED_PLUGINS
              value: "emqx_management,emqx_recon,emqx_retainer,emqx_dashboard"

```

**Comment by tigercl** [comment]
@vborcea Sorry for later response, and I failed to reproduce the problem what you said in my cluster environment. Maybe you can provide the complete profile so that we can troubleshoot the problems.

**Comment by mariusstaicu** [comment]
I can confirm that starting from 3.0.1 onwards, even in 3.1.1 the LWM message is not sent regardless whether the client disconnects gracefully or not. Related issue: https://github.com/emqx/emqx/issues/2564.

**Comment by mariusstaicu** [comment]
If, instead using lwm topic, I use $SYS/brokers/${node}/clients//${clientId}/disconnected, then I receive all disconnect events, regardless of type.

**Comment by mononym** [comment]
This bug was closed as of the 3.2 release as specified here: https://github.com/emqx/emqx/issues/2564#issuecomment-511259704
